packages/api/amplify_api_dart/lib/src/graphql/factories/graphql_request_factory.dart

```diff
@@ -98,8 +108,35 @@ class GraphQLRequestFactory {
     // build inputs based on request operation
     switch (operation) {
       case GraphQLRequestOperation.get:
-        upperOutput = r'($id: ID!)';
-        lowerOutput = r'(id: $id)';
+        final indexes = schema.indexes;
+        final modelIndex =
+            indexes?.firstWhereOrNull((index) => index.name == null);
+        if (modelIndex != null) {
+          // custom index(es), e.g.
+          // upperOutput: ($customId: ID!, name: String!)
+          // lowerOutput: (customId: $customId, name: $name)
+          final bUpperOutput = StringBuffer(_openParen);
+          final bLowerOutput = StringBuffer(_openParen);
+          for (final field in modelIndex.fields) {
+            final isId = field == modelIndex.fields.first;
+            // Assume type is `String` if not specified as `ID`.
```

Comment:
callout: want to check if other types can be used for index, then might have to map all these out